asmrun/fail.c

```diff
@@ -121,11 +121,21 @@ void caml_failwith (char const *msg)
   caml_raise_with_string((value) caml_exn_Failure, msg);
 }
 
+void caml_failwith_value (value msg)
```

Comment:
It looks fishy to me that an OCaml-value-taking primitive would have no `CAMLfoo` ceremony, but I followed the style of the existing `void caml_raise_sys_error(value msg)` declaration a few lines below:

```c

 void caml_raise_end_of_file(void)
 {
   caml_raise_constant((value) caml_exn_End_of_file);
}
```